src/main_thread/decrypt/find_key_system.ts

```diff
@@ -397,3 +396,35 @@ export default function getMediaKeySystemAccess(
     }
   }
 }
+/**
+ * Test a key system configuration, resolves with the MediaKeySystemAccess
+ * or reject if the key system is unsupported.
+ * @param {string} keyType - The KeySystem string to test (ex: com.microsoft.playready.recommendation)
+ * @param {Array.<MediaKeySystemMediaCapability>} keySystemConfigurations - Configurations for this keySystem
+ * @returns Promise resolving with the MediaKeySystemAccess. Rejects if unsupported.
+ */
+export async function testOneKeySystem(
+  keyType: string,
+  keySystemConfigurations: MediaKeySystemConfiguration[],
+) {
+  const keySystemAccess = await eme.requestMediaKeySystemAccess(
+    keyType,
+    keySystemConfigurations,
+  );
+
+  if (!canRelyOnRequestMediaKeySystemAccess(keyType)) {
+    try {
+      const mediaKeys = await keySystemAccess.createMediaKeys();
+      const session = mediaKeys.createSession();
+      // this is just and an example initData to test if the CDM is capable of generating a request
+      const playReadyHeader =
```

Comment:
I feel like that hardcoded string could be put elsewhere than inside this function.

Also are the `LUI_URL` and `LA_URL` mandatory?

People could find it weird to have a canal-plus URL in their code, even if we know this isn't for nefarious purpose, just it was already there for the content we tested on.
So either it's optional and we remove it, either it's not and we could put example.com or something instead.

Reply:
`LUI_URL` and `LA_URL` elements are definitely optional (as stated in [https://learn.microsoft.com/en-us/playready/specifications/playready-header-specification](url)) and should be removed here.

Reply:
Thanks I will remove it. Do you know if I have to update the checksum as the data changed ?
From the documentation I understand that the checksum should be the same because the KID did not changed. I can also remove it because it seems to be optional.

Reply:
The CHECKSUM doesn't seem to be required (from https://learn.microsoft.com/en-us/playready/specifications/playready-header-specification#362-format) 

And from https://learn.microsoft.com/en-us/playready/specifications/playready-header-specification#5-key-checksum-algorithm:

> The checksum algorithm in the PlayReady Header is intended to protect against mismatched keys. In the early days of DRM, songs were encrypted with incorrectly labeled keys. This resulted in white noise being played back when the songs were decrypted. And if the songs were played back loud enough, the playback equipment was destroyed. With the checksum, the content key can be verified as the key that was used to encrypt the file. The algorithm works as follows:
> [...]
> For an ALGID value set to "AESCTR", the 16-byte Key ID is encrypted with a 16-byte AES content key using ECB mode. The first 8 bytes of the buffer is extracted and base64 encoded.

It seems to be only based on keys so OK here?

Very interesting that it seems to be to avoid breaking devices (and ears?) due to badly decrypted audio :D

Reply:
The CHECKSUM element depends on KEYID and CONTENTKEY values exclusively.